### PR TITLE
Fail on index error

### PR DESCRIPTION
**Pull request by johandahlberg**
This changes the behavior of CheckQC to emit a fatal error if there are significantly over-represented indexes. This means that the exit code of the program will not be 0 when this happens.

### CODE REVIEW

**Review by matrulda — APPROVED** [review]
Looks good to me! :+1:

**Comment by matrulda** [comment]
Do you want @b97pla to look at it too or can I merge, @johandahlberg ?

**Comment by johandahlberg** [comment]
Wait with merging a second. I have a fix for #78 incoming.

**Review by matrulda — APPROVED** [review]
Looks good to me! :+1:

File: checkQC/handlers/unidentified_index_handler.py [hunk 1]
```diff
@@ -211,34 +264,50 @@ def __eq__(self, other):
...
 
-    @staticmethod
-    def get_complementary_sequence(seq):
-        conv_table = {'A': 'T', 'T': 'A', 'G': 'C', 'C': 'G', 'N': 'N', '+': '+'}
-        comp_seq = ''
-        for c in seq:
-            comp_seq += conv_table[c]
-        return comp_seq
+        for samplesheet_index in self.samplesheet_dict.keys():
+            if index in samplesheet_index:
```
Comment (b97pla) [inline checkQC/handlers/unidentified_index_handler.py]: perhaps you want to ensure that you are not matching a partial index here? i.e. something like
`if index in samplesheet_index.split('+'):`

**Review by b97pla — APPROVED** [review]
:+1:

File: checkQC/handlers/unidentified_index_handler.py (hunk 1 above)
Comment (johandahlberg) [reply checkQC/handlers/unidentified_index_handler.py]: I've added a change to fix this, in the commit f741a80 have a look at what you think.

Comment (b97pla) [reply checkQC/handlers/unidentified_index_handler.py]: looks good to me, do you want me to merge?

**Comment by johandahlberg** [comment]
Go ahead and merge.

**Pull request merged**